src/main/java/ee/tuleva/onboarding/comparisons/FundComparisonCalculatorService.java

```diff
@@ -1,154 +1,36 @@
 package ee.tuleva.onboarding.comparisons;
 
 import ee.tuleva.onboarding.auth.principal.Person;
-import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
-import ee.tuleva.onboarding.comparisons.fundvalue.FundValueProvider;
 import ee.tuleva.onboarding.comparisons.fundvalue.retrieval.EPIFundValueRetriever;
 import ee.tuleva.onboarding.comparisons.fundvalue.retrieval.WorldIndexValueRetriever;
 import ee.tuleva.onboarding.comparisons.overview.AccountOverview;
 import ee.tuleva.onboarding.comparisons.overview.AccountOverviewProvider;
-import ee.tuleva.onboarding.comparisons.overview.Transaction;
+import ee.tuleva.onboarding.comparisons.returns.RateOfReturnCalculator;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.decampo.xirr.Xirr;
 import org.springframework.stereotype.Service;
 
-import java.math.BigDecimal;
-import java.math.MathContext;
 import java.time.Instant;
-import java.time.ZoneId;
-import java.time.ZoneOffset;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-
-import static java.math.BigDecimal.ZERO;
-import static java.util.stream.Collectors.toList;
 
 @Service
 @RequiredArgsConstructor
 @Slf4j
 public class FundComparisonCalculatorService {
 
     private final AccountOverviewProvider accountOverviewProvider;
-    private final FundValueProvider fundValueProvider;
-
-    private static final int OUTPUT_SCALE = 4;
+    private final RateOfReturnCalculator rateOfReturnCalculator;
 
     public FundComparison calculateComparison(Person person, Instant startTime, Integer pillar) {
         AccountOverview overview = accountOverviewProvider.getAccountOverview(person, startTime, pillar);
-        return calculateForAccountOverview(overview);
-    }
 
-    private FundComparison calculateForAccountOverview(AccountOverview accountOverview) {
-        double actualRateOfReturn = getRateOfReturn(accountOverview);
-        double estonianAverageRateOfReturn = getRateOfReturn(accountOverview, EPIFundValueRetriever.KEY);
-        double marketAverageRateOfReturn = getRateOfReturn(accountOverview, WorldIndexValueRetriever.KEY);
+        double actualRateOfReturn = rateOfReturnCalculator.getRateOfReturn(overview);
```

Comment:
Could we prevent the duplication by having something along the lines of:
```java
getReturn.execute(overview);
```

Reply:
Also, should we add a key for second pillar and pass that in rather than rely on the signature?

Reply:
Will dig deeper in the next PR